pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -290,4 +302,52 @@ public static Map<String, String> getOptionsFromJson(JsonNode request, String op
   public static Map<String, String> getOptionsFromString(String optionStr) {
     return Splitter.on(';').omitEmptyStrings().trimResults().withKeyValueSeparator('=').split(optionStr);
   }
+
+  public static Set<String> getTableNames(SqlNode sqlNode) {
```

Comment:
Just a question, We got rid of a similar function before in this [PR](https://github.com/apache/pinot/pull/10873/files#diff-5bdf53db08d67e2823f300f3db3af05dc704f4ef5319299cc3233f9b55aba059L150). Are we trying to implement it again?

Reply:
Yes, this logic is used in `pinot-java-client`, which depends on `pinot-common` not `pinot-query-planner`.

So the table name extraction logic there only works for v1 query and some simple v2 queries.

There are 3 possible options to solve this:
1. Let `pinot-java-client` depends `pinot-query-planner`;
2. Move some query compilation logics from `pinot-query-planner` to `pinot-common`, for `SqlNode` -> `RelNode`, but it requires the full initialization of Calcite PlannerContext with FrameworkConfig, Prepare.CatalogReader, RelDataTypeFactory, HepProgram.
3. Create a separated getTableName logic to `pinot-common`

Reply:
I think, we removed this function because it failed to get table names for some complex queries. 
I am good with this method.

Reply:
@xiangfu0 i think the major cause of removing this was maintaining the overhead of maintaining this function as we kept on finding edge cases where we were not extracting table names out. Though I see we are pushing to a random broker now so should be good. cc @walterddr what are your thoughts on bringing this method back?

Reply:
@Jackie-Jiang @xiangfu0 ^